### ISSUE TITLE
Clarify Definition of "Leave Game" method

### ISSUE THREAD

**Issue by frozenfrank** [softwareconstruction240/softwareconstruction#69]
## Overview
Last semester, there was disagreement and ambiguity in the definition of "leave game" which was left to the students to figure out.  Some behaviors simply weren't defined which caused some confusion (clearly, it was still manageable). Unless we are intentionally trying to introduce students to the complexities in software design around _ambiguous and unclear specifications_, it may be worth clarifying the specs to resolve this issue. Then this class can be more of a model of what software development _should be_ even students often will encounter ambiguity in the workplace.

Can we explicitly discuss this ambiguity in the specs by showing alternative possible interpretations, even if we refrain from officially embracing one of them over the others?

## Discussion
For the definition of “leaving the game,” there has been some debate among the professors about the defined behavior for the function:
* Like tournament rules, some say that logging out should cause you to automatically lose the game (abandonment)
* Like family rules, others say that leaving the game just leaves an empty place at the table for someone else to step in and finish the game.

Last semester, they tolerated both interpretations of the definition. I’ll see if they want to be more clear about this issue, or at least explicitly say that they tolerate both positions this semester.

**Comment by tannerhs** [comment]
I think either way is fine because in reality, people might choose to implement this either way in a production environment.  Family rules leaves the benefit of allowing sporadic play as you get a moment, while tournament rules allows you to play a game in one sitting with someone in an alternate location.

**Comment by leesjensen** [comment]
I agree that this is confusing, but the majority vote was to leave it confusing and unspecified. The opinion is that being to specific removed the ability for the student to create a solution that they desired. Deciding who and how a game is won doesn't seem like something you want confusion about, but I lost this fight.
